src/webgpu/capability_info.ts

```diff
@@ -87,135 +87,1083 @@ export const kGeneratableErrorScopeFilters = kErrorScopeFilters.filter(
-  'rgba32sint':            [        true,         false,     false,        ,        ,          ,      true,          ,          ,               'sint',              16,             ,              ,                          16,                                4],
-  'rgba32float':           [        true,         false,     false,        ,        ,          ,      true,          ,          , 'unfilterable-float',              16,             ,              ,                          16,                                4],
-} as const);
-/* prettier-ignore */
-const kTexFmtInfoHeader =  ['renderable', 'multisample', 'resolve', 'color', 'depth', 'stencil', 'storage', 'copySrc', 'copyDst',         'sampleType', 'bytesPerBlock', 'blockWidth', 'blockHeight', 'renderTargetPixelByteCost', 'renderTargetComponentAlignment',                  'feature',       'baseFormat'] as const;
-const kSizedDepthStencilFormatInfo = /* prettier-ignore */ makeTable(kTexFmtInfoHeader,
-                           [        true,          true,     false,   false,        ,          ,     false,          ,          ,                     ,                ,            1,             1,                   undefined,                        undefined,                           ,          undefined] as const, {
-  'depth32float':          [            ,              ,          ,        ,    true,     false,          ,      true,     false,              'depth',               4],
-  'depth16unorm':          [            ,              ,          ,        ,    true,     false,          ,      true,      true,              'depth',               2],
-  'stencil8':              [            ,              ,          ,        ,   false,      true,          ,      true,      true,               'uint',               1],
+/**
+ * Defaults applied to all tables automatically. Used only inside `tableWithDefaults`.
+ * This ensures keys are never missing, always explicitly `undefined`.
+ *
+ * All top-level keys must be defined here, or they won't be exposed.
+ */
+const kUniversalDefaults = {
+  blockWidth: undefined,
+  blockHeight: undefined,
+  color: undefined,
+  depth: undefined,
+  stencil: undefined,
+  colorRender: undefined,
+  multisample: undefined,
+  feature: undefined,
+  baseFormat: undefined,
+
+  sampleType: undefined,
+  copySrc: undefined,
+  copyDst: undefined,
+  bytesPerBlock: undefined,
+  renderable: false,
+  renderTargetPixelByteCost: undefined,
+  renderTargetComponentAlignment: undefined,
+
+  // IMPORTANT:
+  // Add new top-level keys both here and in TextureFormatInfo_TypeCheck.
+} as const;
+/**
+ * Takes `table` and applies `defaults` to every row, i.e. for each row,
+ * `{ ... kUniversalDefaults, ...defaults, ...row }`.
+ * This only operates at the first level; it doesn't support defaults in nested objects.
+ */
+function tableWithDefaults<Defaults extends {}, Table extends { readonly [K: string]: {} }>({
```

Comment:
This is only for texture formats, so maybe `textureTableWithDefaults`? Or maybe it's possible to generalize this more and pass in like a templated `UniveralDefaults` also? Not sure if that's possible/worth the effort though... So maybe just a rename is good enough?

For naming also, `kUniversalDefaults`, maybe `kUniversalFormatDefaults`?

Reply:
Good point, originally I was moving all this stuff into another file and then I realized it was too much for one PR. I'll update the names.